src/pudl/output_diff/cli.py

```diff
@@ -0,0 +1,403 @@
+        self.root_path = root_path
+        if root_path.startswith("gs://"):
+            self.fs = fsspec.filesystem("gcs", project=gcs_project)
+        else:
+            self.fs = fsspec.filesystem("file")
+
+    def list_files(self) -> dict[str, str]:
+        """Returns dict mapping from basenames to full paths."""
+        # TODO(rousik): check if root_path is an actual directory.
+        if self.root_path.endswith(".sqlite"):
+            return {"sqlite": self.root_path}
+        else:
+            return {
+                Path(fpath).name: fpath
+                for fpath in self.fs.glob(self.root_path + "/*")
+                if self.match_filetype(fpath)
+            }
+
+    def match_filetype(self, fpath: str) -> bool:
+        """Returns true if file should be considered for comparison."""
+        return any([fpath.endswith(ft) for ft in self.FILE_TYPES])
+
+    def get_engine(self, fname: str):
+        """Returns sqlalchemy engine for reading contents of fname."""
+        return create_engine_remote(self.fs, f"{self.root_path}/{fname}")
+
+    def get_inspector(self, fname: str):
+        """Returns sqlalchemy inspector for analyzing contents of fname."""
+        return inspect(self.get_engine(fname))
+
+    def get_row_count(self, fname: str, table_name: str) -> int:
+        """Returns number of rows contained within the table."""
+        if not re.compile(r"^[a-zA-Z0-9_]+$").match(table_name):
+            raise ValueError("table_name is not SQL safe")
+        # Note that because of the above check, CWE-89 SQL injection issue
+        # is no longer possible below.
+        # Because we get the table names from sqlite metadata, these table
+        # names should be safe by definition also.
+        return (
+            self.get_engine(fname)
+            .execute(text(f"SELECT COUNT(*) FROM {table_name}"))  # nosec
+            .scalar()
+        )
+
+    def get_rows_as_df(self, fname: str, table_name: str) -> pd.DataFrame:
+        """Returns sqlite table contents as pandas DataFrame."""
+        con = self.get_engine(fname)
+        return pd.concat(
+            [df for df in pd.read_sql_table(table_name, con, chunksize=100_000)]
+        )
+
+
+class OutputPair:
+    """Represents pair of output.
+
+    Baseline and an experiment.
+    """
+
+    def __init__(
+        self, baseline: OutputBundle, experiment: OutputBundle, args: argparse.Namespace
+    ):
+        """Create new instance with given baseline and experiment."""
+        self.baseline = baseline
+        self.experiment = experiment
+        self.args = args
+
+    def compare_files(self) -> ListDiff:
+        """Produces GenericDiff that contains file basenames."""
+        baseline_files = set(self.baseline.list_files())
+        experiment_files = set(self.experiment.list_files())
+        return ListDiff.from_sets("File", baseline_files, experiment_files)
+
+    def compare_sqlite_databases(self, file_diff: ListDiff) -> list[CategoryDiff]:
```

Comment:
You mentioned wanting to rework the depth flags above. Was your plan to configure which categories to diff and pass that into this method?